### PR TITLE
first pass at costars batch upload form

### LINKED ISSUE #66: Form to upload CSVs of all COSTARS contracts
Related to #61

### PR DESCRIPTION
**Pull request by hamhands**
### What changed
- added first pass of batch costars upload form
### Issues
- closes #66 
### Still needed
- [ ] clarification on form tokens
- [ ] thought around nuking existing costars contracts, updating contract base with upload

### CODE REVIEW

File: purchasing/conductor/forms.py [hunk 1]
```diff
@@ -30,3 +31,16 @@ class NoteForm(Form):
...
+class FileUpload(Form):
+    upload = FileField('datafile', validators=[
+        FileRequired(),
+        FileAllowed(['xls', 'xlsx'], '.xls and .xlsx files only')
+    ])
+    password = PasswordField('Password',
+                                validators=[DataRequired()])
+
+    def validate_password(form, field):
+        current_password = FileUploadPassword.query.first()
```
Comment (bsmithgall) [inline purchasing/conductor/forms.py]: `FileUploadPassword` isn't defined anywhere.

File: purchasing/conductor/views.py [hunk 2]
```diff
@@ -267,3 +267,15 @@ def assign(contract_id, flow_id, user_id):
...
     return redirect(url_for('conductor.index'))
+
+@blueprint.route('/upload_new', methods=['GET', 'POST'])
+def upload():
+    form = FileUpload()
+    if form.validate_on_submit():
+        _file = request.files.get('upload')
+        filename = secure_filename(_file.filename)
+        filepath = os.path.join(current_app.config.get('UPLOAD_FOLDER'), filename)
+        _file.save(filepath)
```
Comment (bsmithgall) [inline purchasing/conductor/views.py:282]: How do we actually process the data?

**Comment by bsmithgall** [comment]
Left a few inline comments @hamhands -- the big thing is that this change doesn't actually do anything to the file once it is uploaded. On old Wex, we had a few methods that actually handled the data processing so that will need to be supplemented here.

**Comment by hamhands** [comment]
@bsmithgall yeah that's what the first still needed refers to. That's going to take some thinking through tomorrow. Thanks for the comments!

**Comment by bsmithgall** [comment]
What do you mean by "form tokens"?

**Comment by hamhands** [comment]
in each of the form templates there's a jinja thing that says something about form tokens.

Ex: {{ note_form.csrf_token() }}

**Comment by bsmithgall** [comment]
That's related to [CSRF](http://wtforms.readthedocs.org/en/latest/csrf.html)

**Comment by bsmithgall** [comment]
Unrelated to the actual data processing/upload

**Comment by hamhands** [comment]
Ah ok, good to know.

**Comment by bsmithgall** [comment]
@hamhands I'm closing this due to the closing of #62. You should rebase against master and re-submit a PR against master.

**Pull request closed without merging**